src/IssueViz.Security/HotspotsList/ViewModels/HotspotsControlViewModel.cs

```diff
@@ -65,22 +73,49 @@ public HotspotsControlViewModel(IHotspotsStore hotspotsStore, ICommand navigateC
             readonlyObservableHotspotsCollection = allHotspots;
             readonlyObservableHotspotsCollection.CollectionChanged += HotspotsStore_CollectionChanged;
 
-            NavigateCommand = navigateCommand;
-            UpdateHotspotsList(allHotspots);
+            UpdateHotspotsList(allHotspots, Array.Empty<IAnalysisIssueVisualization>());
+
+            SetCommands(hotspotsStore, locationNavigator);
+        }
+
+        private void SetCommands(IHotspotsStore hotspotsStore, ILocationNavigator locationNavigator)
+        {
+            NavigateCommand = new DelegateCommand(parameter =>
+            {
+                var selectedHotspot = (IHotspotViewModel) parameter;
+                locationNavigator.TryNavigate(selectedHotspot.Hotspot);
+            }, parameter => parameter is IHotspotViewModel);
+
+            DeleteCommand = new DelegateCommand(parameter =>
+            {
+                var selectedHotspot = (IHotspotViewModel) parameter;
+                hotspotsStore.Delete(selectedHotspot.Hotspot);
+            }, parameter => parameter is IHotspotViewModel);
         }
 
-        private void UpdateHotspotsList(IEnumerable<IAnalysisIssueVisualization> hotspots)
+        private void UpdateHotspotsList(IEnumerable<IAnalysisIssueVisualization> addedHotspots, IEnumerable<IAnalysisIssueVisualization> deletedHotspots)
         {
-            foreach (var addedHotspot in hotspots)
+            foreach (var addedHotspot in addedHotspots)
             {
                 Hotspots.Add(new HotspotViewModel(addedHotspot));
             }
+
+            var viewModelsToDelete = Hotspots.Where(x => deletedHotspots.Contains(x.Hotspot)).ToList();
```

Comment:
I'm guessing the list needs to be reified here because the hotspots collection is being mutated below.
This might be worth calling out in a comment since it's not entirely obvious - I sometimes add a comment like that, but not always.

I was going to say that since the collection is only used within this method and then discarded, it makes sense to use the cheapest data structure that will do the job i.e. an array instead of a list. However, we don't know if the implementation of ToArray is more or less efficient that that of ToList, so there's no point.